lib/augmentConfig.mjs

```diff
@@ -362,6 +389,34 @@ async function addPluginFunctions(config, { quietDeprecationWarnings }) {
 	return config;
 }
 
+/**
+ *
+ * @param {StylelintConfig} config
+ * @returns {Promise<StylelintConfig>}
+ */
+async function getProcessors(config) {
+	if (!config.processors) {
+		return config;
+	}
+
+	const processors = config.processors.map(async (processorLookup) => {
+		let processorImport;
+
+		if (typeof processorLookup === 'string') {
+			processorImport = await dynamicImport(processorLookup);
+			processorImport = processorImport.default || processorImport;
+		} else {
+			processorImport = processorLookup;
+		}
+
+		return typeof processorImport === 'function' ? await processorImport() : processorImport;
+	});
+
+	config.processors = await Promise.all(processors);
```

Comment:
[suggest] I want to keep `config.processors` as an original value to make debugging easier. So, what about saving processor functions in an internal state like plugins?

https://github.com/stylelint/stylelint/blob/d3351244957e76945241e4b091d0e745b35835ab/lib/augmentConfig.mjs#L387

E.g.

```js
config._processorFunctions = await Promise.all(processors);
```

Reply:
Good point on keeping processors as original value! 
I added:
```js
                _processorFunctions?: {
                        [processorName: string]: Omit<ReturnType<Processor>, 'name'>;
                }; // Internal use only
```
https://github.com/YuanboXue-Amber/stylelint/blob/985bdf1a7f7cd7999867f97153ca4cc440dbb643/types/stylelint/index.d.ts#L121-L123